receiver/statsdreceiver/config.go

```diff
@@ -39,7 +39,7 @@ func (c *Config) Validate() error {
 		}
 
 		switch eachMap.StatsdType {
-		case protocol.TimingTypeName, protocol.TimingAltTypeName, protocol.HistogramTypeName:
+		case protocol.TimingTypeName, protocol.TimingAltTypeName, protocol.HistogramTypeName, protocol.CounterTypeName, protocol.GaugeTypeName:
```

Comment:
Is this an expected change? before this, `protocol.CounterTypeName` & `protocol.GaugeTypeName` would produce an error. After this change they will no longer do so.

Pinging code owners @jmacd & @dmitryax for confirmation here

Reply:
This doesn't look right. Why would a linter complain when there's a default branch? 
The default branch is correct but the error message is misleading. "statsd_type is not a supported mapping" could be replaced with "statsd_type is not a supported mapping for histogram and timing metrics".

I see this change was applied in another PR, so it's pointless to review here? https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24046

Reply:
> This doesn't look right. Why would a linter complain when there's a default branch?

This is a feature of the exhaustive linter: `ExplicitExhaustiveSwitchFlag` https://pkg.go.dev/github.com/nishanths/exhaustive#pkg-constants

Reply:
> I see this change was applied in another PR, so it's pointless to review here? https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24046

Yea sorry about that, I didn't see that other PR